wsproto/events.py

```diff
@@ -140,10 +114,12 @@ class RejectConnection(Event):
 
     """
 
-    _fields = ["headers", "has_body", "status_code"]
-    _defaults = {"headers": list, "has_body": False, "status_code": 400}
+    status_code: int = 400
+    headers: Headers = field(default_factory=list)
+    has_body: bool = False
```

Comment:
Do we have to always specify the type even for simple bool and int variables?

Reply:
Hmm, I think so - I can't see anything in the docs to suggest it can be infered.